### ISSUE TITLE
Ts code not working after bun build

### ISSUE THREAD

**Issue by pjlunardelli** [oven-sh/bun#9037]
### What version of Bun is running?

1.0.28+705638470

### What platform is your computer?

Darwin 23.3.0 arm64 arm

### What steps can reproduce the bug?

```
bun build index.ts --target=bun --outfile index.js 
bun run index.js
```

1. Run pushSSHPublicKey:

```js
import {
  EC2InstanceConnectClient,
  SendSSHPublicKeyCommand,
} from "@aws-sdk/client-ec2-instance-connect";

const { AWS_REGION, INSTANCE_ID, BASTION_USER, AVAILABILITY_ZONE } =
  process.env;

const pushSSHPublicKey = async (publicKey: string) => {
  const client = new EC2InstanceConnectClient({ region: AWS_REGION });

  const command = new SendSSHPublicKeyCommand({
    InstanceId: INSTANCE_ID,
    InstanceOSUser: BASTION_USER,
    SSHPublicKey: publicKey,
    AvailabilityZone: AVAILABILITY_ZONE,
  });

  try {
    await client.send(command);

    console.log("SSH public key successfully pushed to the instance.");
    return true;
  } catch (error) {
    console.error("Failed to push SSH public key:", error);
    return false;
  }
};

export { pushSSHPublicKey };
```

.env of:
```
AWS_REGION=us-west-2 (is my case, but probably not relevant)
INSTANCE_ID (can't share)
BASTION_USER (can't share)
AVAILABILITY_ZONE (can't share)
AWS_PROFILE=xxxx (This is important, if this is not defined, it works as expected), I am using AWS identity center on my profile
```

.aws/config contents:
```
[default]
sso_account_id = (can't share)
sso_region = (can't share)
sso_start_url=(can't share)
sso_role_name = (can't share)
region = us-west-2
[profile (can't share)]
sso_session = (can't share)
sso_account_id = (can't share)
sso_role_name = (can't share)
[sso-session (can't share)]
sso_start_url = (can't share)
sso_region = (can't share)
sso_registration_scopes = (can't share)
```

### What is the expected behavior?

Same result from
bun run index.ts 

And:
```
bun build index.ts --target=bun --outfile index.js 
bun run index.js
```

### What do you see instead?

When I run bun run index.ts, the code works as expected.

When I run:
```
bun build index.ts --target=bun --outfile index.js 
bun run index.js
```

I get the error bellow:

```js
26731 | // node_modules/@aws-sdk/client-sso/dist-es/SSOClient.js
26732 | class SSOClient extends Client {
26733 |   constructor(...[configuration]) {
26734 |     const _config_0 = getRuntimeConfig4(configuration || {});
                              ^
TypeError: getRuntimeConfig4 is not a function. (In 'getRuntimeConfig4(configuration || {})', 'getRuntimeConfig4' is undefined)
      at new SSOClient (/Users/pedrolunardelli/code/generateSecureBastionConnection/index.js:26734:23)
      at /Users/pedrolunardelli/code/generateSecureBastionConnection/index.js:27108:30

Error: 31541 |   try {
31542 |     const { publicKey, privateKey } = generateKeyPair();
31543 |     await prepareDir();
31544 |     const keyUpdated = await pushSSHPublicKey(publicKey);
31545 |     if (!keyUpdated)
31546 |       throw new Error("Failed to update the SSH public key.");
                    ^
error: Failed to update the SSH public key.
      at /Users/pedrolunardelli/code/generateSecureBastionConnection/index.js:31546:13
```

For some context of index.js (generated by bun build index.ts --target=bun --outfile index.js ):
```js
  // node_modules/@aws-sdk/client-sso/dist-es/runtimeConfig.js
var getRuntimeConfig4;
var init_runtimeConfig2 = __esm(() => {
  init_package2();
  init_dist_es37();
  init_dist_es41();
  init_dist_es11();
  init_dist_es42();
  init_dist_es31();
  init_dist_es14();
  init_dist_es28();
  init_dist_es43();
  init_dist_es20();
  init_runtimeConfig_shared2();
  init_dist_es30();
  init_dist_es44();
  init_dist_es30();
  getRuntimeConfig4 = (config7) => {
    emitWarningIfUnsupportedVersion(process.version);
    const defaultsMode = resolveDefaultsModeConfig(config7);
    const defaultConfigProvider = () => defaultsMode().then(loadConfigsForDefaultMode);
    const clientSharedValues = getRuntimeConfig3(config7);
    emitWarningIfUnsupportedVersion3(process.version);
    return {
      ...clientSharedValues,
      ...config7,
      runtime: "node",
      defaultsMode,
      bodyLengthChecker: config7?.bodyLengthChecker ?? calculateBodyLength,
      defaultUserAgentProvider: config7?.defaultUserAgentProvider ?? defaultUserAgent({ serviceId: clientSharedValues.serviceId, clientVersion: package_default3.version }),
      maxAttempts: config7?.maxAttempts ?? loadConfig(NODE_MAX_ATTEMPT_CONFIG_OPTIONS),
      region: config7?.region ?? loadConfig(NODE_REGION_CONFIG_OPTIONS, NODE_REGION_CONFIG_FILE_OPTIONS),
      requestHandler: config7?.requestHandler ?? new NodeHttpHandler(defaultConfigProvider),
      retryMode: config7?.retryMode ?? loadConfig({
        ...NODE_RETRY_MODE_CONFIG_OPTIONS,
        default: async () => (await defaultConfigProvider()).retryMode || DEFAULT_RETRY_MODE
      }),
      sha256: config7?.sha256 ?? Hash.bind(null, "sha256"),
      streamCollector: config7?.streamCollector ?? streamCollector,
      useDualstackEndpoint: config7?.useDualstackEndpoint ?? loadConfig(NODE_USE_DUALSTACK_ENDPOINT_CONFIG_OPTIONS),
      useFipsEndpoint: config7?.useFipsEndpoint ?? loadConfig(NODE_USE_FIPS_ENDPOINT_CONFIG_OPTIONS)
    };
  };
});

class SSOClient extends Client {
  constructor(...[configuration]) {
    const _config_0 = getRuntimeConfig4(configuration || {});
    const _config_1 = resolveClientEndpointParameters3(_config_0);
    const _config_2 = resolveRegionConfig(_config_1);
    const _config_3 = resolveEndpointConfig(_config_2);
    const _config_4 = resolveRetryConfig(_config_3);
    const _config_5 = resolveHostHeaderConfig(_config_4);
    const _config_6 = resolveUserAgentConfig(_config_5);
    const _config_7 = resolveHttpAuthSchemeConfig3(_config_6);
    const _config_8 = resolveRuntimeExtensions2(_config_7, configuration?.extensions || []);
    super(_config_8);
    this.config = _config_8;
    this.middlewareStack.use(getRetryPlugin(this.config));
    this.middlewareStack.use(getContentLengthPlugin(this.config));
    this.middlewareStack.use(getHostHeaderPlugin(this.config));
    this.middlewareStack.use(getLoggerPlugin(this.config));
    this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
    this.middlewareStack.use(getUserAgentPlugin(this.config));
    this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
      httpAuthSchemeParametersProvider: this.getDefaultHttpAuthSchemeParametersProvider(),
      identityProviderConfigProvider: this.getIdentityProviderConfigProvider()
    }));
    this.middlewareStack.use(getHttpSigningPlugin(this.config));
  }
  destroy() {
    super.destroy();
  }
  getDefaultHttpAuthSchemeParametersProvider() {
    return defaultSSOHttpAuthSchemeParametersProvider;
  }
  getIdentityProviderConfigProvider() {
    return async (config7) => new DefaultIdentityProviderConfig({
      "aws.auth#sigv4": config7.credentials
    });
  }
}
```

**Comment by pjlunardelli** [comment]
Can I help with any additional information?

**Comment by uanid** [comment]
I have same problem. 

<img width="1032" alt="image" src="https://github.com/oven-sh/bun/assets/35264628/c772f3cc-2120-4783-82e4-486627f45436">

**Comment by ilijapuaca** [comment]
Same issue here using `@aws-sdk/client-secrets-manager`

**Comment by fitzn** [comment]
Same issue here using `@aws-sdk/client-secrets-manager`.

**Comment by Jarred-Sumner** [comment]
This is fixed in Bun v1.1.21

To get the release early:
```sh
bun upgrade --canary
```

**Comment by fitzn** [comment]
Thank you all!
